engine/api/workflow/dao_run.go

```diff
@@ -335,3 +336,50 @@ func nextRunNumber(db gorp.SqlExecutor, w *sdk.Workflow) (int64, error) {
 	log.Debug("nextRunNumber> %s/%s %d", w.ProjectKey, w.Name, i)
 	return int64(i), nil
 }
+
+// PurgeWorkflowRun mark all worfklow run to delete
+func PurgeWorkflowRun(db gorp.SqlExecutor, wf sdk.Workflow, branch string) error {
+	if wf.HistoryLength == 0 {
+		log.Warning("PurgeWorkflowRun> history length equals 0, skipping purge")
+		return nil
+	}
+
+	queryUpdate := `UPDATE workflow_run SET to_delete = true WHERE workflow_run.id IN ( %s )`
+	args := []interface{}{wf.ID}
+	if branch == "" {
+		queryUpdate = fmt.Sprintf(queryUpdate, `SELECT workflow_run.id FROM workflow_run
+			LEFT JOIN workflow_run_tag ON workflow_run.id = workflow_run_tag.workflow_run_id
+			WHERE workflow_run.workflow_id = $1
+			AND (workflow_run_tag.tag IS NULL
+				OR (workflow_run_tag.tag = 'git.branch' AND (workflow_run_tag.value IS NULL OR workflow_run_tag.value = ''))
```

Comment:
i'm not sure i like to hard code to delete workflow run depending on a specific tag.
If i'm running workflow with svn, it won't be possible to delete by .svn.branch ?